### PR TITLE
Task: Follow up for #14431 Carbon 

### PR DESCRIPTION
**Pull request by kingjia90**
Resolves https://github.com/pimcore/pimcore/pull/14431/files?w=1#r1121530982

### CODE REVIEW

**Review by jdreesen — APPROVED** [review]

File: models/Document/Editable/Date.php [hunk 1]
```diff
@@ -27,7 +28,7 @@ class Date extends Model\Document\Editable implements EditmodeDataInterface
      *
      * @internal
      *
-     * @var \Carbon\Carbon|null
+     * @var Carbon|null
```
Comment (blankse) [inline models/Document/Editable/Date.php:31]: if the $date property can be \DateTimeInterface and not Carbon it should be changed here
```suggestion
     * @var \DateTimeInterface|null
```

Comment (blankse) [reply models/Document/Editable/Date.php:31]: And also for `getData()` and `getDate()`

File: models/Document/Editable/Date.php [hunk 2]
```diff
@@ -73,7 +74,7 @@ public function getDataEditmode() /** : mixed */
     public function frontend()
     {
         if ($this->date instanceof \DateTimeInterface) {
```
Comment (blankse) [inline models/Document/Editable/Date.php]: But I don't see a case that the date can be set an DateTimeInterface without Carbon. I think it is save to change it directly here:
```suggestion
        if ($this->date instanceof Carbon) {
```

File: models/Document/Editable/Date.php [hunk 3]
```diff
@@ -73,7 +74,7 @@ public function getDataEditmode() /** : mixed */
...
-            if (isset($this->config['outputFormat']) && $this->config['outputFormat']) {
+            if (isset($this->config['outputFormat']) && $this->config['outputFormat'] && $this->date instanceof Carbon) {
```
Comment (blankse) [inline models/Document/Editable/Date.php]:
```suggestion
            if (isset($this->config['outputFormat']) && $this->config['outputFormat']) {
```

File: models/Document/Editable/Date.php (hunk 2 above)
Comment (kingjia90) [reply models/Document/Editable/Date.php]: mhh, that's right, everything was and will (in 11) be `Carbon`  anyways

File: models/Document/Editable/Date.php (hunk 1 above)
Comment (martineiber) [reply models/Document/Editable/Date.php:31]: No, this should stay with Carbon.

**Review by martineiber — APPROVED** [review]

**Pull request merged**